src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -176,6 +103,16 @@ public void AddPeer(IPEndPoint endPoint, IPAddress source)
             this.peers.TryAdd(peerToAdd.Endpoint, peerToAdd);
         }
 
+        public void RemovePeer(IPEndPoint endPoint, IPAddress source)
```

Comment:
I find it odd that you need to add `IPAddress source`  this should be inside IPEndPoint no?

Reply:
Yes, it does not make sense.

Reply:
Also inheritdoc

Reply:
yes agreed it feels weird, but this is only to go through the same code path as for AddPeer, for symmetry

Reply:
Okay, remove it

Reply:
then you have to ensure that both methods rely on IPV6, which I haven't found a better way to enforce

Reply:
there is an extension method for that, something like ensureipv6 or something